airflow/providers/amazon/aws/operators/emr.py

```diff
@@ -617,6 +617,14 @@ def execute(self, context: Context) -> str | None:
                     job_id=self.job_id,
                     aws_conn_id=self.aws_conn_id,
                     waiter_delay=self.poll_interval,
+                    waiter_max_attempts=self.max_polling_attempts,
+                )
+                if self.max_polling_attempts
```

Comment:
Wouldn't this be a better solution?
```
waiter_max_attempts=self.max_polling_attempts if self.max_polling_attempts else 600
```
You can write a test where you provide a `max_polling_attempts`, and one where you don't, and assert that the correct value is used.

Reply:
@syedahsn  I felt that using hardcoded values (such as 600) in the code can be problematic for maintenance. Especially when such a value is referenced in multiple places, if a change is necessary, finding and modifying all the reference points increases the likelihood of making errors. Therefore, I chose to avoid defining defaults directly in the code.

Reply:
Hmm that's a good point. In that case, maybe you can edit the Trigger code to accept `None` (i.e. `waiter_max_attempts: int | None = 600`) as well, and do (I also included the `waiter_delay` param as well)


```
waiter_max_attempts = waiter_max_attempts or 600
waiter_delay = waiter_delay or 30
```
You would then be able to pass just pass `max_polling_attempts` as is, and the Trigger will handle the `None` case. What do you think about this approach?

Reply:
@syedahsn 
What you just mentioned was my initial approach. 
However, after analyzing the code, I found that modifying only the input type of `[EmrStepSensorTrigger](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/emr.py#L213)` results in a discrepancy with its parent class, violating the Liskov Substitution Principle (LSP). If we were to modify the parent class, `[AwsBaseWaiterTrigger](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/triggers/base.py#L31)`, we would also need to change the type of `waiter_max_attempts` in the `[async_await](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/utils/waiter_with_logging.py#L88)` utility that accepts this input. This would require handling the exception when `waiter_max_attempts` is None. 
(https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/utils/waiter_with_logging.py#L119)
Additionally, this function is widely used across AWS operators, and we cannot predict what other side effects it might cause.

Considering these factors, I believe the current code modification, while not entirely clean, is the best approach.

Reply:
Alright I guess making those changes would go beyond the scope of this PR, so I agree. Great work with the deep dive!